docs/content/customization/generic/encryption-at-rest.md

```diff
@@ -0,0 +1,60 @@
++++
+title = "Encryption At REST"
++++
+
+`encryptionAtRest` variable enables encrypting kubernetes resources at REST using provided encryption provider.
+When this variable is set, kuberntetes secrets and configmaps are encrypted before writing them at `etcd`.
+
+If the `encryptionAtRest` property is not specified, then
+the customization will be skipped. The secrets and configmaps will not be stored as encrypted in `etcd`.
+
+We support following encryption providers
+
+- aescbc
+- secretbox
+
+More information about encryption at REST: [Encrypting Confidential Data at Rest
+](https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/)
+
```

Comment:
Let's add some details about the key generation, otherwise a user might not understand how a key is provided to the encryption config.

Reply:
added details: https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/659/commits/4046a9400ec03420fa98ea7f6da8ac8c3f996d9d